src/ncp/ncp_uart.cpp

```diff
@@ -176,7 +176,12 @@ void NcpUart::EncodeAndSendToUart(void)
 
     if (len > 0)
     {
-        otPlatUartSend(mUartBuffer.GetBuffer(), len);
+        if (otPlatUartSend(mUartBuffer.GetBuffer(), len) != kThreadError_None)
+        {
+            mUartBuffer.Clear();
+
+            mUartSendTask.Post();
+        }
```

Comment:
I'd remove the buffer clear. I'd also keep track of the returned error and assert (if we aren't in release mode). We shouldn't ever get to this point in normal behavior.